samples/file-encryptor/enclave/openssl_src/encryptor.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openssl/aes.h>
+#include <openssl/evp.h>
+#include <string.h>
+
+#include "aes_context.h" // Must be included before common/encryptor.h
+#include "common/encryptor.h"
+#include "common/trace.h"
+
+ecall_dispatcher::ecall_dispatcher() : m_encrypt(true), m_header(NULL)
+{
+}
+
+int ecall_dispatcher::initialize(
+    bool encrypt,
+    const char* password,
+    size_t password_len,
+    encryption_header_t* header)
+{
+    int ret = 1;
+    TRACE_ENCLAVE(
+        "ecall_dispatcher::initialize : %s request",
+        encrypt ? "encrypting" : "decrypting");
+
+    if (header == NULL)
+    {
+        TRACE_ENCLAVE("initialize() failed as NULL was passed in place of "
+                      "(encryption_header_t *)");
+        goto exit;
+    }
+
+    m_password = string(password, password + password_len);
+    m_encrypt = encrypt;
+    m_header = header;
+
+    memset((void*)m_encryption_key, 0, ENCRYPTION_KEY_SIZE_IN_BYTES);
+
+    ret = process_encryption_header(header);
+    if (ret != 0)
+    {
+        TRACE_ENCLAVE("process_encryption_header failed with %d", ret);
+        goto exit;
+    }
+
+    // init iv
+    memcpy(m_operating_iv, m_header->salt, IV_SIZE);
+
+    // initialize aes context
+    m_aescontext = (struct aes_context*) malloc(sizeof(struct aes_context));
+
+    if (m_aescontext == NULL)
+    {
+        ret = 1;
+        TRACE_ENCLAVE("allocate m_aescontext failed with %d", ret);
+        goto exit;
+    }
+
+    if (!(m_aescontext->ctx = EVP_CIPHER_CTX_new()))
+    {
+        TRACE_ENCLAVE("Encryption cipher context instantiation failed");
+        goto exit;
+    }
+
+    // set aes key
+    if (encrypt)
+    {
+        if (!(ret = EVP_EncryptInit_ex(
+                  m_aescontext->ctx,
+                  EVP_aes_256_cbc(),
+                  NULL,
+                  m_encryption_key,
+                  m_operating_iv)))
+        {
+            TRACE_ENCLAVE("EVP_EncryptInit_ex failed with %d", ret);
+            goto exit;
+        }
+    }
+    else
+    {
+        if (!(ret = EVP_DecryptInit_ex(
+                  m_aescontext->ctx,
+                  EVP_aes_256_cbc(),
+                  NULL,
+                  m_encryption_key,
+                  m_operating_iv)))
+        {
+            TRACE_ENCLAVE("EVP_DecryptInit_ex failed with %d", ret);
+            goto exit;
+        }
+    }
+    
+    EVP_CIPHER_CTX_set_padding(
+        m_aescontext->ctx,
+        0); // host application takes care of padding
+
+    EVP_CIPHER_CTX_set_key_length(
+        m_aescontext->ctx, ENCRYPTION_KEY_SIZE_IN_BYTES);
+
+    ret = 0;
+exit:
+    return ret;
+}
+
+int ecall_dispatcher::encrypt_block(
+    bool encrypt,
+    unsigned char* input_buf,
+    unsigned char* output_buf,
```

Comment:
```suggestion
    unsigned char* input_buffer,
    unsigned char* output_buffer,
```
avoid abbreviations, per https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md#Coding-Conventions

Reply:
implemented